doc/source/data/getting-started.rst

```diff
@@ -1,229 +1,86 @@
-batches of records in the Dataset using the given function. The function must return
-a batch of records. You are allowed to filter or add additional records to the batch,
-which will change the size of the Dataset.
-
-Transformations are executed *eagerly* and block until the operation is finished.
-
-.. code-block:: python
-
-    def transform_batch(df: pandas.DataFrame) -> pd.DataFrame:
-        return df.applymap(lambda x: x * 2)
+---------------------
 
-    ds = ray.data.range_arrow(10000)
-    ds = ds.map_batches(transform_batch, batch_format="pandas")
-    # -> Map Progress: 100%|████████████████████| 200/200 [00:00<00:00, 1927.62it/s]
-    ds.take(5)
-    # -> [{'value': 0}, {'value': 2}, ...]
+Once you have a Dataset, you can transform it by applying a user-defined function and produce another Dataset.
+Under the hood the transformation is executed in parrallel for performance at scale.
 
-The batch format can be specified using ``batch_format`` option, which defaults to "native",
-meaning pandas format for Arrow-compatible batches, and Python lists for other types. You
-can also specify explicitly "arrow" or "pandas" to force a conversion to that batch format.
-The batch size can also be chosen. If not given, the batch size will default to entire blocks.
+.. literalinclude:: ./doc_code/quick_start.py
+   :language: python
+   :start-after: __data_transform_begin__
+   :end-before: __data_transform_end__
 
 .. tip::
 
-    Datasets also provides the convenience methods ``map``, ``flat_map``, and ``filter``, which are not vectorized (slower than ``map_batches``), but may be useful for development.
-
-By default, transformations are executed using Ray tasks.
-For transformations that require setup, specify ``compute=ray.data.ActorPoolStrategy(min, max)`` and Ray will use an autoscaling actor pool of ``min`` to ``max`` actors to execute your transforms.
-For a fixed-size actor pool, specify ``ActorPoolStrategy(n, n)``.
-The following is an end-to-end example of reading, transforming, and saving batch inference results using Ray Data:
-
-.. code-block:: python
-
-    from ray.data import ActorPoolStrategy
-
-    # Example of GPU batch inference on an ImageNet model.
-    def preprocess(images: List[bytes]) -> List[bytes]:
-        return images
-
-    class BatchInferModel:
-        def __init__(self):
-            self.model = ImageNetModel()
-        def __call__(self, batch: pd.DataFrame) -> pd.DataFrame:
-            return self.model(batch)
+    Datasets also provides the convenience transformation methods ``map``, ``flat_map``, and ``filter``, which are not vectorized (slower than ``map_batches``), but may be useful for development.
 
-    ds = ray.data.read_binary_files("s3://bucket/image-dir")
+The transformation is composable. You can further apply transformations on the output Dataset, forming
+a chain of transformations. Ray Dataset will optimize the execution by fusing transformations when appropriate.
```

Comment:
Branding: We always use "Ray Datasets" when referring to the library.
```suggestion
a chain of transformations. Ray Datasets will optimize the execution by fusing transformations when appropriate.
```

Reply:
Also given that this stage fusion only happens for lazy datasets and pipelines, we might not want to mention stage fusion in the Getting Started guide.